### DISCUSSION TITLE
Content Types missing after update

### DISCUSSION THREAD

**Discussion by anthrogan** [strapi/strapi#7473] [Legacy Topics]
I have a strapi service running in kubernetes, it connects to a mongo db on another server. 
When I update/restart the pod I lose all of the content types from the admin panel but no actual data, I was under the impression strapi was able to recreate its types after a restart from the database? 

I'm assuming this is not the case, is there any suggested data persistence approach? Is it better to inject the code needed for the content types on the build stage, or should I create some persistent storage for the strapi "config"?

Also for anyone looking, if you just recreate your content types with the same name, all the data is immediately collected form the db, so there's no data loss as such just config loss.

**Comment by MattieBelt** [comment]
Hey @anthrogan,
When editing/creating content types, you are saving the models in files. Upon restart your (docker) container will be _reset_ and thus changes are reverted. If you are using k8s locally you need to setup some volume sharing between the containerized application and your local source code. If using an hosting provider you best can use **Git** as an version control-system to track your file changes.

Strapi's general development flow is:
> Development - Develop your Strapi application locally on your host machine, then push file changes into source control
Staging - Deploy changes from source control to a "production-like" environment for testing
Production - If no other changes are needed, deploy into production
Repeat as needed, it is recommended that you properly version and test your application as you go

_https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#why-can-t-i-create-or-update-content-types-in-production-staging_

**Reply by anthrogan** [reply]
Thanks for the reply. I like the idea of the git approach, i guess it means I can pull the changes from a repo as I rebuild in k8. My current process is just build a container, then run a bash script to build a strapi instance and connect it to a database

Cheers!

**Reply by MattieBelt** [reply]
Yes you should pull the changes from the repo on container build and then run in your (bash) script.
1) Install packages `yarn` or `npm install`
2) Build admin panel `yarn build` or `npm run build`
3) Start Strapi `yarn start` or `npm start`

You can configure your database config in the config files and use environment variables to set them.
_https://strapi.io/documentation/v3.x/concepts/configurations.html#environment-variables_
